src/NexusMods.App.UI/Controls/ModInfo/ModFiles/ViewModFilesView.axaml.cs

```diff
@@ -0,0 +1,135 @@
+
+namespace NexusMods.App.UI.Controls.ModInfo.ModFiles;
+using ModFileNode = TreeNodeVM<IFileTreeNodeViewModel, GamePath>;
+
+public partial class ViewModFilesView : ReactiveUserControl<IModFilesViewModel>
+{
+    public ViewModFilesView()
+    {
+        InitializeComponent();
+        this.WhenActivated(disposables =>
+        {
+            this.WhenAnyValue(view => view.ViewModel)
+                .WhereNotNull()
+                .Do(PopulateFromViewModel)
+                .Subscribe()
+                .DisposeWith(disposables);
+        });
+    }
+
+    private void PopulateFromViewModel(IModFilesViewModel vm)
+    {
+        var source = CreateTreeSource(ViewModel!.Items);
+        source.SortBy(source.Columns[0], ListSortDirection.Ascending);
+        ModFilesTreeDataGrid.Source = source;
+            
+        // Hide Stack Panel
+        MultiLocationStackPanel.IsVisible = ViewModel.RootCount > 1;
+        SingleLocationStackPanel.IsVisible = ViewModel.RootCount <= 1;
+
+        SingleLocationText.Text = ViewModel.PrimaryRootLocation;
+        MultiLocationCountText.Text = $"{ViewModel.RootCount}";
+    }
+
+    private static HierarchicalTreeDataGridSource<ModFileNode> CreateTreeSource(
+        ReadOnlyObservableCollection<ModFileNode> treeRoots)
+    {
+        return new HierarchicalTreeDataGridSource<ModFileNode>(treeRoots)
+        {
+            Columns =
+            {
+                new HierarchicalExpanderColumn<ModFileNode>(
+                    new TemplateColumn<ModFileNode>(
+                        Language.Helpers_GenerateHeader_NAME,
+                        new FuncDataTemplate<ModFileNode>((node, _) =>
+                            {
+                                // This should never be null but can be during rapid resize, due to
+                                // virtualization shenanigans. Think this is a control bug, but well, gotta work with what we have.
+                                // ReSharper disable once ConditionIsAlwaysTrueOrFalseAccordingToNullableAPIContract
+                                if (node == null)
+                                    return new Control();
+
+                                var view = new FileTreeNodeView();
+                                view.DataContext = node.Item;
+                                
+                                // This is a 'hack' which allows us to receive events from the wrapping 'ModFileNode'
+                                // and transfer it into the child IFileTreeNodeViewModel.
+                                
+                                // Given that we may reuse the `FileTreeNodeView` for other views in the future, 
+                                // which may not necessarily need the wrapping `ModFileNode` to get info on when it's
+                                // expanded or not expanded, this does not seem unreasonable to do.
+                                ((IActivatableViewModel)node.Item).WhenActivated(d =>
```

Comment:
I still think this should not be in the codebehind.
Move this to the ViewModel and try using the xaml based DataTemplate method we discussed instead of creating a template from code

Reply:
I just haven't gotten around to this yet, nothing more to say.

Reply:
Moved to #1048